presto-main/src/test/java/io/prestosql/sql/planner/TestDynamicFilter.java

```diff
@@ -254,8 +261,8 @@ public void testNestedDynamicFiltersRemoval()
                                 INNER,
                                 ImmutableList.of(equiJoinClause("ORDERS_CK", "ORDERS_CK6")),
                                 ImmutableMap.of("ORDERS_CK", "ORDERS_CK6"),
-                                Optional.empty(),
-                                tableScan("orders", ImmutableMap.of("ORDERS_CK", "clerk")),
+                                anyTree(
```

Comment:
`testNestedDynamicFiltersRemoval` supposedly tests that the filter got removed
but, with `anyTree` here, we can't really know, can we?

Reply:
In `JoinMatcher#matchDynamicFilters` we perform a traversal on left side of join looking for dynamic filters in FilterNode. So in all the tests in this file even though `anyTree` is given on probe side, `matchDynamicFilters` should fail when expectations around presence of dynamic filter are violated.
But if it's clearer to explicitly check for FilterNode, I'll make the change to add that.